packages/@react-aria/utils/src/scrollIntoView.ts

```diff
@@ -10,32 +10,67 @@
  * governing permissions and limitations under the License.
  */
 
+import {getScrollParent} from './';
+
+let isScrollPrevented = false;
+
+export function setScrollPrevented(value: boolean) {
+  isScrollPrevented = value;
+}
+
+interface ScrollOpts {
+  scrollView: HTMLElement
+}
+
+/**
+ * Scrolls `scrollView` so that `element` is visible.
+ * Similar to `element.scrollIntoView({block: 'nearest'})` (not supported in Edge),
+ * but doesn't affect parents above `scrollView`.
+ * @deprecated
+ */
+export function scrollIntoView(scrollView: HTMLElement, element: HTMLElement);
 /**
  * Scrolls `scrollView` so that `element` is visible.
  * Similar to `element.scrollIntoView({block: 'nearest'})` (not supported in Edge),
  * but doesn't affect parents above `scrollView`.
  */
-export function scrollIntoView(scrollView: HTMLElement, element: HTMLElement) {
-  let offsetX = relativeOffset(scrollView, element, 'left');
-  let offsetY = relativeOffset(scrollView, element, 'top');
-  let width = element.offsetWidth;
-  let height = element.offsetHeight;
+export function scrollIntoView(element: HTMLElement, opts: ScrollOpts);
+export function scrollIntoView(element: HTMLElement, opts: HTMLElement | ScrollOpts) {
```

Comment:
See discussion [here](https://github.com/adobe/react-spectrum/pull/3899#discussion_r1066458214) for background on this change. This will also give us some wiggle room in the future if/when we expand the scroll behavior to mimic things like `block: center` instead

Reply:
shouldn't need the override since you've specified that the second arg can be either

Reply:
I think the overrides is overkill, it's simple enough to just allow either of them, and we can use the one that is more readable in our code. the other can just be a shortcut 🤷🏻 
less confusion about which one is deprecated too

Reply:
fair enough, pushed the update